protocol/test/core/ascend-zaps-test.ts

```diff
@@ -0,0 +1,557 @@
+          await alan.getAddress(),
+          swapAmount,
+          vault.address,
+          singleSwap.assetOut,
+          amountCalculated
+        );
+
+      expect(await TEMPLE.balanceOf(balanceVaultAddress)).to.be.eq(
+        beforeBalance.add(singleSwap.amount)
+      );
+      expect(await DAI.balanceOf(alan.address)).to.be.gt(beforeDAIBalance);
+    });
+  });
+
+  describe('Redeem vault token to temple & Check interest accrued', async () => {
+    let balance: BigNumber;
+    let swapAmount: BigNumber;
+    let singleSwap: IBalancerVault.SingleSwapStruct;
+    let fundMng: IBalancerVault.FundManagementStruct;
+    let deadline: number;
+
+    beforeEach(async () => {
+      balance = await vault.balanceOf(alan.address);
+      swapAmount = balance.div(2);
+      singleSwap = {
+        poolId:
+          '0x177f00928ebc2b222af50a7595af7e7ef0d86a7b000200000000000000000380',
+        kind: 0,
+        assetIn: TEMPLE.address,
+        assetOut: constants.tokens.DAI,
+        amount: swapAmount,
+        userData: '0x',
+      };
+      fundMng = {
+        sender: ascendZaps.address,
+        fromInternalBalance: false,
+        recipient: alan.address,
+        toInternalBalance: false,
+      };
+      deadline = Math.floor(new Date().getTime() / 1000) + 10 * 60;
+    });
+
+    it('only redeem when vault can exit', async () => {
+      const canExit = await vault.canExit();
+      expect(canExit).to.be.eq(false);
+
+      await expect(
+        ascendZaps.redeemVaultTokenToTemple(vault.address)
+      ).to.be.revertedWith('CanNotExitVault');
+
+      await advanceTimeAndBlock(periodDuration);
+
+      await ascendZaps.redeemVaultTokenToTemple(vault.address);
+    });
+
+    it('only owner can redeem', async () => {
+      await advanceTimeAndBlock(periodDuration);
+
+      await expect(
+        ascendZaps.connect(alan).redeemVaultTokenToTemple(vault.address)
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+
+    it('Redeem vault token to temple & Check interest accrued', async () => {
+      await advanceTimeAndBlock(periodDuration);
+      const res = await vault.inEnterExitWindow();
+      const cycleNum = res.cycleNumber;
+      expect(cycleNum).to.be.eq(1);
+
+      let interestAccrued = await ascendZaps.checkInterestAccrued(
+        vault.address
+      );
+      expect(interestAccrued).to.be.eq(0);
+
+      await vault.connect(alan).approve(ascendZaps.address, swapAmount);
+      await ascendZaps
+        .connect(alan)
+        .exitVaultEarly(
+          swapAmount,
+          vault.address,
+          singleSwap,
+          fundMng,
+          0,
+          deadline
+        );
+
+      interestAccrued = await ascendZaps.checkInterestAccrued(vault.address);
+      expect(interestAccrued).to.be.eq(0);
+
+      await ascendZaps.redeemVaultTokenToTemple(vault.address);
+      interestAccrued = await ascendZaps.checkInterestAccrued(vault.address);
+      expect(interestAccrued).to.be.eq(0);
```

Comment:
simulate extra transactions to make sure interest is accrued. should get some hints in other test files.